File: graphene-reader/src/main/java/net/iponweb/disthene/reader/GrapheneReaderApplication.java
package net.iponweb.disthene.reader;

import com.graphene.reader.store.IndexProperty;
import com.graphene.reader.store.key.selector.KeySelectorProperty;
import net.iponweb.disthene.reader.config.*;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.cassandra.CassandraAutoConfiguration;
import org.springframework.boot.autoconfigure.gson.GsonAutoConfiguration;
import org.springframework.boot.context.properties.EnableConfigurationProperties;

@SpringBootApplication(scanBasePackages = {"com.graphene.reader", "net.iponweb.disthene.reader"})
@EnableAutoConfiguration(exclude = {CassandraAutoConfiguration.class, GsonAutoConfiguration.class})
@EnableConfigurationProperties({
  IndexProperty.class,
  StoreConfiguration.class,
  GrapheneReaderProperties.class,
  ThrottlingConfiguration.class,
  ReaderConfiguration.class,
  KeySelectorProperty.class
})
public class GrapheneReaderApplication {
  public static void main(String[] args) {
    SpringApplication.run(GrapheneReaderApplication.class, args);
  }
}

File: graphene-reader/src/main/java/net/iponweb/disthene/reader/GrapheneReaderApplication.kt
package net.iponweb.disthene.reader

import net.iponweb.disthene.reader.config.IndexConfiguration
import org.apache.log4j.BasicConfigurator
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.autoconfigure.cassandra.CassandraAutoConfiguration
import org.springframework.boot.autoconfigure.gson.GsonAutoConfiguration
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.runApplication

@SpringBootApplication
@EnableAutoConfiguration(exclude = [
    CassandraAutoConfiguration::class,
    GsonAutoConfiguration::class
])
@EnableConfigurationProperties(IndexConfiguration::class)
open class GrapheneReaderApplication

fun main(args: Array<String>) {
    runApplication<GrapheneReaderApplication>(*args)

    BasicConfigurator.configure()
}
